omnigibson/robots/holonomic_base_robot.py

```diff
@@ -0,0 +1,308 @@
+            obs_modalities (str or list of str): Observation modalities to use for this robot. Default is ["rgb", "proprio"].
+                Valid options are "all", or a list containing any subset of omnigibson.sensors.ALL_SENSOR_MODALITIES.
+                Note: If @sensor_config explicitly specifies `modalities` for a given sensor class, it will
+                    override any values specified from @obs_modalities!
+            proprio_obs (str or list of str): proprioception observation key(s) to use for generating proprioceptive
+                observations. If str, should be exactly "default" -- this results in the default proprioception
+                observations being used, as defined by self.default_proprio_obs. See self._get_proprioception_dict
+                for valid key choices
+            sensor_config (None or dict): nested dictionary mapping sensor class name(s) to specific sensor
+                configurations for this object. This will override any default values specified by this class.
+            kwargs (dict): Additional keyword arguments that are used for other super() calls from subclasses, allowing
+                for flexible compositions of various object subclasses (e.g.: Robot is USDObject + ControllableObject).
+        """
+        self._world_base_fixed_joint_prim = None
+
+        # Call super() method
+        super().__init__(
+            name=name,
+            relative_prim_path=relative_prim_path,
+            scale=scale,
+            visible=visible,
+            fixed_base=fixed_base,
+            visual_only=visual_only,
+            self_collisions=self_collisions,
+            load_config=load_config,
+            abilities=abilities,
+            control_freq=control_freq,
+            controller_config=controller_config,
+            action_type=action_type,
+            action_normalize=action_normalize,
+            reset_joint_pos=reset_joint_pos,
+            obs_modalities=obs_modalities,
+            proprio_obs=proprio_obs,
+            sensor_config=sensor_config,
+            **kwargs,
+        )
+
+    def _post_load(self):
+        super()._post_load()
+
+        self._world_base_fixed_joint_prim = lazy.omni.isaac.core.utils.prims.get_prim_at_path(
+            f"{self.prim_path}/rootJoint"
+        )
+        position, orientation = self.get_position_orientation()
+        # Set the world-to-base fixed joint to be at the robot's current pose
+        self._world_base_fixed_joint_prim.GetAttribute("physics:localPos0").Set(tuple(position))
+        self._world_base_fixed_joint_prim.GetAttribute("physics:localRot0").Set(
+            lazy.pxr.Gf.Quatf(*orientation[[3, 0, 1, 2]].tolist())
+        )
+
+    def _initialize(self):
+        super()._initialize()
+
+        # Set the max velocity for the base joints
+        self.joints["base_footprint_x_joint"].max_velocity = m.MAX_LINEAR_VELOCITY
```

Comment:
should we assert first that we have correct naming schemes? i assume that these joint names are expected to be standard across all holonomic robots

Reply:
Done!